### DISCUSSION TITLE
Where to get datarefs and commands

### DISCUSSION THREAD

**Discussion by fatflash** [mauer/xmidictrl#26] [Support]
Hi there,
First let me thank you for your work, I have installed and used X-touch Mini with your plugin and operated my TBM 900 to perfection!!!
Now I am trying to expand operations to my Phenom 300, and was wondering if you could share some tips on where to find Datarefs and commands other than the SASL file and Keyboard commands.
Many thanks again for your help.
Al.

**Comment by mauer** [comment]
Hi,

Many thanks for the kind words. I love the TBM 900, as well and cannot wait to have it in X-Plane 12 again.

Finding the correct Datarefs and Commands can be quite a pain. There are a few things you can do:

1. Use the [DataRefTool](https://github.com/leecbaker/datareftool). It tries to find and display all Datarefs and Commands available. You can filter by datarefs that have been changed recently.
2. Have a look at the OBJ files of the aircraft for Datarefs
3. Contact the developer and ask (I had to do that for a couple of planes).

Hope that helps,
Marco

**Reply by fatflash** [reply]
Hi Marco,
Thanks for your reply.
Eventually I have figured it out. I have already posted in the discussions/Community aircraft profiles the ones I have been working on, along with the SVG stencils.

One thing to notice is that my XP12 doesn't seem to like the plugin very much, as it always closes with a crash when your plugin is installed.
Are you aware of this?
I have a mac mini M2 pro with the latest sonoma and XP12.

Thanks.
Al.
